WordPress/Classes/ViewRelated/Blog/Blog Details/Detail Header/BlogDetailHeaderView.swift

```diff
@@ -53,16 +55,11 @@ class BlogDetailHeaderView: UIView {
     }
 
     @objc func refreshIconImage() {
-        if let blog = blog,
-            blog.hasIcon == true {
-            titleView.siteIconView.imageView.downloadSiteIcon(for: blog)
```

Comment:
Probably related to the Photon thing you mentioned, this function was used to display the site icon, and is now replaced with a plain URL built using `WPImageURLHelper`. What do you think switching the image loading back to how it's implemented in the original `downloadSiteIcon` function?

Reply:
I think we should remove `downloadSiteIcon`:
- It is broken for private sites
- It uses `AlamofireImage` that does too much on the main thread, and I ideally want to remove, like I already done with a couple of other image loading system we had. The new system is an `ImageDownloader` and every part of the app should use it.
- wp.com uses Photon inconsistently. I'm not sure what the pattern is. We might need to rethink how these URLs are constructed. I don't think it kept up with the times.

Reply:
Let's get this PR merged and continue iterating on the download logic later. This code exists in `trunk`, so I'm not changing it in this PR but just extracting it to a separate view.

It works better than the production version where for me half of the site icons are not loading.

Reply:
Hey, @crazytonyli, I think I found a good compromise to avoid messing with the download logic too much.

- Move `UIImage.optimizedURL` to `SiteIconViewModel` so we no longer extending the native types, which is good. `SiteIconViewModel` now uses the exact same logic to create the URL as the production version `downloadSiteIcon`
- Update `CachedAsyncImage` to support `MediaHost`, which was the second half of the `downloadSiteIcon` code. 

The download logic is now largely equivalent with the production version. The only difference is that it uses our `ImageDownloader` instead of `AlamofireImage`. The new downloader does much less on the main thread. It invokes `MediaRequestAuthenticator` in the background and creates and starts the request in the background.

I also slightly changed `CachedAsyncImage`. It wasn't doing a good job using the memory cache.

So, we now have a good starting point with the behavior that matches production. The problem is – in production, many of the P2 site icons are failing to load. @jkmassel did a little bit of an investigation and concluded that the app needs to send `Bearer:` for this requests, but I'm not sure that's a complete picture because it already does that. I think there is something wrong again with private sites.